app/src/pages/inside/projectSettingsPageContainer/content/defectTypes/modals/addEditDefectTypeModal/addEditDefectTypeModal.jsx

```diff
@@ -177,7 +181,7 @@ const AddEditDefectTypeModal = ({
       okButton={okButton}
       cancelButton={cancelButton}
       onClose={() => dispatch(hideModalAction())}
-      allowCloseOutside={!dirty}
+      allowCloseOutside={!isFormChanged}
```

Comment:
And this to `color?.toLowerCase() === defectType.color`

Reply:
Okay, maybe I didn't understand something.

But anyway, looks like now we can combine `color?.toLowerCase() === defectType.color` and `dirty` to achieve the expected results. 

allowCloseOutside={!dirty || color?.toLowerCase() === defectType.color}

Reply:
The bug was that `dirty` was not correct.
We don't change the color in the end, but dirty=true.
And `!true && color?.toLowerCase() === defectType.color` =false